webhooks/data-import-cron_test.go

```diff
@@ -0,0 +1,372 @@
+			delete(dict.Spec.Template.Spec.Storage.Resources.Requests, v1.ResourceStorage)
+
+			err := ValidateDataImportCronTemplate(dict)
+			Expect(err).To(MatchError(ContainSubstring("Storage size is missing")))
+		})
+
+		It("should fail if storage size is missing in PVC", func() {
+			dict.Spec.Template.Spec.PVC = pvcSpecFromStorageSpec(dict.Spec.Template.Spec.Storage)
+			dict.Spec.Template.Spec.Storage = nil
+
+			delete(dict.Spec.Template.Spec.PVC.Resources.Requests, v1.ResourceStorage)
+
+			err := ValidateDataImportCronTemplate(dict)
+			Expect(err).To(MatchError(ContainSubstring("PVC size is missing")))
+		})
+
+		It("should fail if storage size is negative in Storage", func() {
+			dict.Spec.Template.Spec.Storage.Resources.Requests[v1.ResourceStorage] = resource.MustParse("-20G")
+
+			err := ValidateDataImportCronTemplate(dict)
+			Expect(err).To(MatchError(ContainSubstring("Storage size can't be equal or less than zero")))
+		})
+
+		It("should fail if storage size is negative in PVC", func() {
+			dict.Spec.Template.Spec.PVC = pvcSpecFromStorageSpec(dict.Spec.Template.Spec.Storage)
+			dict.Spec.Template.Spec.Storage = nil
+
+			dict.Spec.Template.Spec.PVC.Resources.Requests[v1.ResourceStorage] = resource.MustParse("-20G")
+
+			err := ValidateDataImportCronTemplate(dict)
+			Expect(err).To(MatchError(ContainSubstring("PVC size can't be equal or less than zero")))
+		})
+
+		It("should fail if storage class is invalid in Storage", func() {
+			dict.Spec.Template.Spec.Storage.StorageClassName = ptr.To("invalid storage class <>[]{}()--++!@#$%^&*|';:")
+
+			err := ValidateDataImportCronTemplate(dict)
+			Expect(err).To(MatchError(ContainSubstring("invalid storage class name:")))
+		})
+
+		It("should fail if storage class is invalid in PVC", func() {
+			dict.Spec.Template.Spec.PVC = pvcSpecFromStorageSpec(dict.Spec.Template.Spec.Storage)
+			dict.Spec.Template.Spec.Storage = nil
+
+			dict.Spec.Template.Spec.PVC.StorageClassName = ptr.To("invalid storage class <>[]{}()--++!@#$%^&*|';:")
+
+			err := ValidateDataImportCronTemplate(dict)
+			Expect(err).To(MatchError(ContainSubstring("invalid storage class name:")))
+		})
+
+		It("should fail if access mode is invalid in Storage", func() {
+			dict.Spec.Template.Spec.Storage.AccessModes = []v1.PersistentVolumeAccessMode{
+				"unknown-access-mode",
+			}
+
+			err := ValidateDataImportCronTemplate(dict)
+			Expect(err).To(MatchError(ContainSubstring("unsupported value:")))
+		})
+
+		It("should fail if access mode is invalid in PVC", func() {
+			dict.Spec.Template.Spec.PVC = pvcSpecFromStorageSpec(dict.Spec.Template.Spec.Storage)
+			dict.Spec.Template.Spec.Storage = nil
+
+			dict.Spec.Template.Spec.PVC.AccessModes = []v1.PersistentVolumeAccessMode{
+				"unknown-access-mode",
+			}
+
+			err := ValidateDataImportCronTemplate(dict)
+			Expect(err).To(MatchError(ContainSubstring("unsupported value:")))
+		})
+
+		It("should fail if there are multiple access modes in PVC", func() {
```

Comment:
This case is missing for Storage?

Reply:
Can the cases not be split e.g. like other ones?

Reply:
> Can the cases not be split e.g. like other ones?

What do you mean?